faiss/gpu/GpuIndex.h

```diff
@@ -77,11 +77,17 @@ class GpuIndex : public faiss::Index {
     /// as needed
     /// Handles paged adds if the add set is too large; calls addInternal_
     void add(idx_t, const float* x) override;
+    void add(idx_t, const void* x, NumericType numeric_type) override;
 
     /// `x` and `ids` can be resident on the CPU or any GPU; copies are
     /// performed as needed
     /// Handles paged adds if the add set is too large; calls addInternal_
     void add_with_ids(idx_t n, const float* x, const idx_t* ids) override;
+    void add_with_ids(
+            idx_t n,
+            const void* x,
+            NumericType numeric_type,
+            const idx_t* xids) override;
```

Comment:
Maybe we should have taken the opportunity to also make the id sizes parameterizable. There are many use cases where int32 is more appropriate than int64

Reply:
We can change that in a subsequent diff, but I would like to avoid having 3 different add_with_ids implementaitons.